Update brand container to use latest psammead-brand version

Bopchy opened the pull request:
Resolves #1805 

**Overall change:** _Update brand container to use latest psammead-brand component._

**Code changes:**

- _Update psammead-brand dependency to ^4.0.0_
- _Add product, serviceLocalizedName, svgHeight, minWidth, maxWidth, url props to brand container._
- _Add relative service links to service configs._
- _Update tests and snapshots._
- _Update storybook._
---

- [x] I have assigned myself to this PR and the corresponding issues
- [ ] Tests added for new features
- [ ] Test engineer approval
- [ ] I have followed [the merging checklist](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/MERGE_PROCESS.md) and this is ready to merge.

Commit:
Update psammead-brand version.

Commit:
Add links to igbo, yoruba, pidgin and news services.

Commit:
Add svgHeight, minWidth, maxWidth, url props to brand.

Commit:
Update snapshots. Fix eslint issues.

Commit:
Merge branch 'latest' of github.com:bbc/simorgh into 1805-update-brand-component-to-latest-version

sadickisaac commented on src/app/lib/config/services/igbo.js:
```diff
@@ -9,8 +9,10 @@ const igbo = {
   defaultImage: 'https://news.files.bbci.co.uk/ws/img/logos/og/igbo.png',
   defaultImageAltText: 'BBC News Ìgbò',
   brandSVG,
+  svgRatio: brandSVG.ratio,
```
Do we need to add `svgRatio` to config?

Bopchy replied:
We actually don't have to - have removed svgRatio from config. svgRatio is now being retrieved from brandSvg

Commit:
Merge branch 'latest' into 1805-update-brand-component-to-latest-version

Commit:
Remove svgRatio from config. Fix typo in withPageWrapper.jsx. Update snapshots.

Commit:
Merge branch '1805-update-brand-component-to-latest-version' of github.com:bbc/simorgh into 1805-update-brand-component-to-latest-version

Commit:
Merge branch 'latest' of github.com:bbc/simorgh into 1805-update-brand-component-to-latest-version

Bopchy commented on src/app/containers/Brand/index.jsx:
```diff
@@ -3,9 +3,23 @@ import Brand from '@bbc/psammead-brand';
+  const { brandName, brandSVG, link } = useContext(ServiceContext);
+  const svgMaxHeight = 24;
+  const svgMinHeight = 16;
+  const svgRatio = brandSVG && brandSVG.ratio;
```
Added this check because test withPageWrapper.test.jsx was failing, and seeing brandSVG as undefined despite defining it in

sadickisaac replied:
I don't think we need that check either. You could pass `service` to  `<PageWrapperHOC />`. `withPageWrapper` is failing because no service was passed to `PageWrapperHOC` it therefore defaults to the default service config. The `default` config doesn't have `BrandSVG` defined. `configs` are in `lib/services`

Bopchy replied:
Hmmh, the news `service` is currently being passed to `<PageWrapperHOC />` through dataProps,  but (without the check to see if brandSVG is defined) - the test still fails

sadickisaac replied:
You need to pass it on the component `<PageWrapperHOC data={dataProps} service="news" />`

sadickisaac replied:
`PageWrapperHOC` returns `defaultPageWrapper` which expects service as a prop.

Bopchy replied:
👍 Made the change

Commit:
Merge branch 'latest' into 1805-update-brand-component-to-latest-version

Commit:
Merge branch 'latest' of github.com:bbc/simorgh into 1805-update-brand-component-to-latest-version

Commit:
Merge branch 'latest' into 1805-update-brand-component-to-latest-version

Commit:
Merge branch 'latest' into 1805-update-brand-component-to-latest-version

Commit:
Remove brandSVG check. Update snapshot.

Commit:
Merge branch 'latest' of github.com:bbc/simorgh into 1805-update-brand-component-to-latest-version

Commit:
Merge branch '1805-update-brand-component-to-latest-version' of github.com:bbc/simorgh into 1805-update-brand-component-to-latest-version

Commit:
Update snapshot.

Commit:
Merge branch 'latest' of github.com:bbc/simorgh into 1805-update-brand-component-to-latest-version

jamesbhobbs commented on src/app/lib/config/services/news.js:
```diff
@@ -23,6 +23,7 @@ const news = {
   noBylinesPolicy: 'https://www.bbc.com/news/help-41670342#authorexpertise',
   publishingPrinciples: 'https://www.bbc.com/news/help-41670342',
   script: latin,
+  link: 'https://bbc.com/news',
```
needs www. also I'm unsure if we should can/should use .co.uk or .com hardcoded or pass it in conditionally based on the variant we're rendering. Suggest you check with other maintainers.

Bopchy replied:
Checked and @pjlee11 supports having the option of both .co.uk and .com

This will be handled in #1909

Commit:
Add www.

Commit:
Merge branch 'latest' of github.com:bbc/simorgh into 1805-update-brand-component-to-latest-version

Commit:
Merge branch 'latest' of github.com:bbc/simorgh into 1805-update-brand-component-to-latest-version

sadickisaac commented on src/app/lib/config/services/igbo.js:
```diff
@@ -11,6 +11,7 @@ const igbo = {
   brandSVG,
   script: latin,
   fonts: [],
+  link: 'https://www.bbc.com/igbo',
```
Could we change this to something more descriptive. ie. `frontpageUrl`

Commit:
Use frontpageUrl instead of link.

Commit:
Merge branch 'latest' of github.com:bbc/simorgh into 1805-update-brand-component-to-latest-version

DenisBBC commented:
Is it worth to wait until [this](https://github.com/bbc/psammead/pull/609) PR is merged, before moving this forward? Otherwise, we will have to update again the container.

DenisHdz reviewed (commented):
LGTM, but do you think is it worth to wait until [this](https://github.com/bbc/psammead/pull/609) PR is merged, before moving this forward? Otherwise, we will have to update the container again.

Bopchy commented:
Yes, let's wait until it's merged

12 commented on src/app/containers/Brand/index.jsx:
```diff
@@ -3,9 +3,23 @@ import Brand from '@bbc/psammead-brand';
 
 const BrandContainer = () => {
-  const { brandName, brandSVG } = useContext(ServiceContext);
+  const { brandName, brandSVG, frontpageUrl } = useContext(ServiceContext);
```
Just a thought, do we need to define a default `brandSVG` here:

https://github.com/bbc/simorgh/blob/f745b0082b783ebd27fc2eee57bf6ce6138d6ffe/src/app/lib/config/services/index.js#L13-L22

as we have this that sets the default service to `default`:

https://github.com/bbc/simorgh/blob/f745b0082b783ebd27fc2eee57bf6ce6138d6ffe/src/app/contexts/ServiceContext/index.jsx#L19-L21

12 replied:
I realise this isn't a change you've explicitly made, however it now fails more abruptly now when you don't filter out the `default` story. 

(also I don't see the point in having a default value if we're just going to filter it out in storybook 🤷‍♂)

Currently, `latest` throws a failed prop type, however still renders the component, without the SVG.
This branch would now throw a `TypeError` which causes the entire component to fail and not render.

I think this should be addressed before merging

Bopchy replied:
Handled failure in absence of brandSVG by adding a check for brandSVG here - https://github.com/bbc/simorgh/blob/28dec513a1bf6fd2e7043e9eaf63c4b6c53ddc1b/src/app/containers/Brand/index.jsx#L11. 

Regarding if a default brandSVG should be defined - it's been decided it's not necessary, because (with the added check) the component works without an SVG, and the prop errors are currently not an issue

Commit:
Add brandSVG check.
- Added this to handle when there is no brandSVG defined.

Commit:
Merge branch 'latest' of github.com:bbc/simorgh into 1805-update-brand-component-to-latest-version

Commit:
Update package-lock.json

Bopchy commented:
Waiting on https://github.com/bbc/psammead/pull/609 to be merged so that changes are made once

dr3 commented:
This will now pass the URL link as `/news` or `/igbo` etc, to be relative.

Commit:
Use relative urls.
- Use relative urls with origin generated dynamically, instead of hardcoded urls.

Commit:
Update snapshots.

Commit:
Merge branch 'latest' of github.com:bbc/simorgh into 1805-update-brand-component-to-latest-version

Commit:
Merge branch 'latest' of github.com:bbc/simorgh into 1805-update-brand-component-to-latest-version

Bopchy commented:
Now waiting on https://github.com/bbc/psammead/issues/606

UPDATE: Doesn't need to wait on  https://github.com/bbc/psammead/issues/606 as Simorgh will pick up the latest non-major change in the psammead-brand package

Commit:
Merge branch 'latest' of github.com:bbc/simorgh into 1805-update-brand-component-to-latest-version

Commit:
Update brand container Storybook, tests and snapshots.

Bopchy commented on src/app/lib/config/services/news.js:
```diff
@@ -6,6 +6,7 @@ import * as fonts from '@bbc/psammead-styles/fonts';
 const news = {
   articleAuthor: `https://www.facebook.com/bbcnews`,
   brandName: 'BBC News',
+  product: 'BBC News',
```
Added this because Storybook fails on the news service without it. I am not sure if it is required

ryanmccombe replied:
I think the issue is Brand container is now expecting a key called `product` on context, where before it was using `brandName` (the value on the previous line)

You may also have to make this change to the other languages, or have Brand container continue to use `brandName` from the ServiceContext, instead of this new `product` value

Bopchy replied:
Aah, yes. I thought there might be a reason why it was added to other services but not to news.

Commit:
Merge branch 'latest' of github.com:bbc/simorgh into 1805-update-brand-component-to-latest-version

Commit:
Update snapshots.

Commit:
Merge branch 'latest' of github.com:bbc/simorgh into 1805-update-brand-component-to-latest-version

Commit:
Update snapshots.

j-pendlebury reviewed (approved):
LGTM 👍

dr3 reviewed (changes requested):
This change makes the brand not be restricted to 1280px, is it broken in psammead?
![image](https://user-images.githubusercontent.com/11341355/59665011-040dec80-91aa-11e9-9aa8-67ef569082b9.png)

dr3 commented:
This PR is blocked on https://github.com/bbc/psammead/issues/645

Commit:
Merge branch 'latest' of github.com:bbc/simorgh into 1805-update-brand-component-to-latest-version

jamesbhobbs committed:
Merge branch 'latest' into 1805-update-brand-component-to-latest-version

Commit:
Update cypress tests.

Commit:
Merge branch '1805-update-brand-component-to-latest-version' of github.com:bbc/simorgh into 1805-update-brand-component-to-latest-version

Commit:
Update package-lock.json

Commit:
Merge branch 'latest' of github.com:bbc/simorgh into 1805-update-brand-component-to-latest-version

Commit:
Merge branch 'latest' of github.com:bbc/simorgh into 1805-update-brand-component-to-latest-version

Commit:
Update package-lock.json

Commit:
Merge branch 'latest' of github.com:bbc/simorgh into 1805-update-brand-component-to-latest-version

Commit:
Fix failing e2e.

Commit:
Update snapshots.

Commit:
Merge branch 'latest' of github.com:bbc/simorgh into 1805-update-brand-component-to-latest-version

Bopchy commented on cypress/integration/frontpageLayoutSpec.js:
```diff
@@ -23,7 +23,7 @@ Object.keys(services).forEach(index => {
             .should('have.lengthOf', 1)
             .find('a')
-            .should('have.attr', 'href', 'https://www.bbc.co.uk/news') // expect `${config.baseUrl}${serviceConfig.url}` once header hooked up
+            .should('have.attr', 'href', `${serviceConfig.url}`)
```
Using `${serviceConfig.url}` in favour of ` ${config.baseUrl}${serviceConfig.url}` since BrandContainer passes a relative url

ryanmccombe replied:
```suggestion
            .should('have.attr', 'href', serviceConfig.url)
```

dr3 commented on package.json:
```diff
@@ -80,7 +80,7 @@
     "@bbc/gel-foundations": "3.0.0",
     "@bbc/psammead-assets": "^0.1.9",
-    "@bbc/psammead-brand": "^1.0.1",
+    "@bbc/psammead-brand": "^4.0.0",
```
Needs to be ^4.0.1

dr3 commented on src/app/containers/Brand/index.jsx:
```diff
@@ -3,9 +3,26 @@ import Brand from '@bbc/psammead-brand';
 
-  return <Brand brandName={brandName} svg={brandSVG} />;
+  return (
+    <Brand
```
Need to pass `borderTop` or `borderBottom` depending on if its in the header or footer, can be set by passing a prop when calling this container

dr3 commented on src/app/containers/PageHandlers/withPageWrapper.test.jsx:
```diff
@@ -7,11 +7,11 @@ const dataProps = {
   shouldMatchSnapshot(
     `should render correctly`,
-    <PageWrapperHOC {...dataProps} />,
+    <PageWrapperHOC data={dataProps} service="news" />,
```
why this change? I dont think service is an expected prop https://github.com/bbc/simorgh/blob/1805-update-brand-component-to-latest-version/src/app/containers/PageHandlers/withPageWrapper.jsx#L14

Bopchy replied:
Added because `PageWrapperHOC` uses the `default` service, and not the news service specified in dataProps - unless the service is passed to it as a prop

Bopchy replied:
This change isn't needed. Reverted it.

dr3 commented on src/app/containers/Brand/index.jsx:
```diff
@@ -3,9 +3,26 @@ import Brand from '@bbc/psammead-brand';
+      minWidth={minWidth}
+      maxWidth={maxWidth}
+      svg={brandSVG}
+      url={frontpageUrl}
```
Can we just use `/${service}`, saves us adding a new item to the config

Commit:
Update version to 4.0.1. Use service instead of frontpageUrl.

Commit:
Merge branch 'latest' of github.com:bbc/simorgh into 1805-update-brand-component-to-latest-version

Commit:
Update snapshots. Add props isHeader and isFooter.

sadickisaac commented on src/app/containers/Brand/index.jsx:
```diff
@@ -1,11 +1,41 @@
+      minWidth={minWidth}
+      maxWidth={maxWidth}
+      svg={brandSVG}
+      url={`/${service}`}
```
Could we make this a variable.

sadickisaac commented on src/app/containers/Header/__snapshots__/index.test.jsx.snap:
```diff
@@ -83,37 +80,34 @@ exports[`Header should render correctly 1`] = `
       <a
         className="c2 c3"
-        href="https://www.bbc.co.uk/news"
+        href="/undefined"
```
Why are we getting undefined here?

Bopchy replied:
This was because the source of the url wasn't defined in the test stub. This is now defined

Commit:
Merge branch 'latest' into 1805-update-brand-component-to-latest-version

Commit:
Merge branch 'latest' of github.com:bbc/simorgh into 1805-update-brand-component-to-latest-version

Commit:
Update tests. Update snapshots.

Commit:
Merge branch '1805-update-brand-component-to-latest-version' of github.com:bbc/simorgh into 1805-update-brand-component-to-latest-version

Commit:
Merge branch 'latest' of github.com:bbc/simorgh into 1805-update-brand-component-to-latest-version

Commit:
Merge branch 'latest' of github.com:bbc/simorgh into 1805-update-brand-component-to-latest-version

Commit:
Merge branch 'latest' of github.com:bbc/simorgh into 1805-update-brand-component-to-latest-version

ryanmccombe commented on cypress/integration/frontpageLayoutSpec.js:
```diff
@@ -71,7 +71,7 @@ Object.keys(services).forEach(index => {
             .should('have.attr', 'role', 'contentinfo')
             .find('a')
-            .should('have.attr', 'href', 'https://www.bbc.co.uk/news') // expect `${config.baseUrl}${serviceConfig.url}` once footer hooked up
+            .should('have.attr', 'href', `${serviceConfig.url}`)
```
`.should('have.attr', 'href', serviceConfig.url)`

ryanmccombe replied:
```suggestion
            .should('have.attr', 'href', serviceConfig.url)
```

ryanmccombe committed:
Merge branch 'latest' into 1805-update-brand-component-to-latest-version

Commit:
Remove literal syntax

Co-Authored-By: Ryan McCombe <ryanmccombe@gmail.com>

ryanmccombe committed:
Merge branch 'latest' into 1805-update-brand-component-to-latest-version

Commit:
Merge branch 'latest' of github.com:bbc/simorgh into 1805-update-brand-component-to-latest-version

Commit:
Merge branch 'latest' into 1805-update-brand-component-to-latest-version

amywalkerdev committed:
Merge branch 'latest' into 1805-update-brand-component-to-latest-version

Commit:
Merge branch 'latest' into 1805-update-brand-component-to-latest-version

Commit:
Merge branch 'latest' into 1805-update-brand-component-to-latest-version

Commit:
Merge branch 'latest' of github.com:bbc/simorgh into 1805-update-brand-component-to-latest-version

jamesbrumpton committed:
Merge branch 'latest' into 1805-update-brand-component-to-latest-version

Commit:
Merge branch 'latest' into 1805-update-brand-component-to-latest-version

Commit:
Merge branch 'latest' of github.com:bbc/simorgh into 1805-update-brand-component-to-latest-version

Commit:
Update snapshots.

Commit:
Merge branch '1805-update-brand-component-to-latest-version' of github.com:bbc/simorgh into 1805-update-brand-component-to-latest-version

david-boydell commented:
LGTM this is ready for merge

Commit:
Merge branch 'latest' into 1805-update-brand-component-to-latest-version

jroebu14 commented on src/app/containers/Header/index.jsx:
```diff
@@ -3,7 +3,7 @@ import BrandContainer from '../Brand';
 const HeaderContainer = () => (
   <header role="banner">
-    <BrandContainer />
+    <BrandContainer isHeader />
```
I'm a bit late with this one but wouldn't it be better to just do `<BrandContainer borderBottom />` here and `<BrandContainer borderTop />` in the footer and then in `Brand` you can spread the props on the component like

```jsx
    <Brand
      product={product}
      serviceLocalisedName={serviceLocalizedName}
      svgHeight={svgMaxHeight}
      minWidth={minWidth}
      maxWidth={maxWidth}
      svg={brandSVG}
      url={`/${service}`}
      {...props}
    />
```

sadickisaac replied:
@jroebu14 Not too late. Made that change  https://github.com/bbc/simorgh/pull/2146
